tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EnumSuite.scala

```diff
@@ -0,0 +1,317 @@
+package scala.meta.tests.parsers.dotty
+
+import scala.meta.tests.parsers._
+import scala.meta._
+import scala.meta.tests.tokenizers.TokenizerSuite
+
+class EnumSuite extends BaseDottySuite {
+
+  implicit val parseStat: String => Stat = code => templStat(code)(dialects.Dotty)
+
+  /**
+   *  All examples based on dotty documentation:
+   *  https://dotty.epfl.ch/docs/reference/enums/enums.html
+   *  https://dotty.epfl.ch/docs/reference/enums/adts.html
+   */
+  // ---------------------------------
+  // ENUM
+  // ---------------------------------
+
+  val RGCase = Defn.RepeatedEnumCase(Nil, List(tname("R"), tname("G")))
+
+  test("enum") {
+    runTestAssert[Stat]("enum Color { case R, G }")(
+      Defn.Enum(Nil, pname("Color"), Nil, ctor, tpl(List(RGCase)))
+    )
+  }
+
+  test("enum-parametrized") {
+    runTestAssert[Stat]("enum C(i: Int) { case R, G }")(
+      Defn.Enum(Nil, pname("C"), Nil, ctorp(List(tparam("i", "Int"))), tpl(List(RGCase)))
+    )
+  }
+
+  test("enum-extends") {
+    runTestAssert[Stat]("enum C extends T with R { case R, G }")(
+      Defn.Enum(
+        Nil,
+        pname("C"),
+        Nil,
+        ctor,
+        Template(Nil, List(init("T"), init("R")), slf, List(RGCase))
+      )
+    )
+  }
+
+  test("enum-generics") {
+    runTestAssert[Stat]("enum C[X, Y] { case R, G }")(
+      Defn.Enum(Nil, pname("C"), List(pparam("X"), pparam("Y")), ctor, tpl(List(RGCase)))
+    )
+  }
+
+  test("enum-min-one-case") {
+    runTestError("enum Color { }", "Enumerations must contain at least one case")
```

Comment:
I would drop this check, it doesn't help at all and might create errors when we use it in Metals